### PR TITLE
[ENG-2673] Show Add New for button for OSFRegistries

### PR DESCRIPTION
**Pull request by futa-ikeda**
- Ticket: [ENG-2673]
- Feature flag: n/a

## Purpose
- Allow users to create a draft registration to OSF Registries by using the Add New button

## Summary of Changes
- Show Add New button when looking at OSF Registries

## Side Effects
- NA

## QA Notes
- Add New button will be visible on `osf.io/registries/discover` route. This button will take users to `osf.io/registries/osf/new` and users should be able to submit to OSF registries from there

[ENG-2673]: https://openscience.atlassian.net/browse/ENG-2673

### CODE REVIEW

File: lib/registries/addon/components/registries-navbar/component.ts [hunk 1]
```diff
@@ -43,7 +43,7 @@ export default class RegistriesNavbar extends AuthBase {
     @computed('provider.{allowSubmissions,id}')
     get showAddRegistrationButton() {
         if (!this.provider) {
```
Comment (brianjgeiger) [inline lib/registries/addon/components/registries-navbar/component.ts:45]: Could probably just take out this whole branch

**Review by brianjgeiger — CHANGES_REQUESTED** [review]
One item.

**Review by brianjgeiger — DISMISSED** [review]

File: lib/registries/addon/components/registries-navbar/component.ts [hunk 2]
```diff
@@ -43,7 +43,7 @@ export default class RegistriesNavbar extends AuthBase {
...
-            return false;
+            return true;
```
Comment (fabmiz) [inline lib/registries/addon/components/registries-navbar/component.ts:46]: maybe `defaultProviderId = 'osf'`

File: lib/registries/addon/components/registries-navbar/component.ts [hunk 3]
```diff
@@ -32,6 +32,7 @@ export default class RegistriesNavbar extends AuthBase {
     @service features!: Features;
 
     provider?: RegistrationProviderModel;
+    defaultProviderId: string = 'osf';
```
Comment (brianjgeiger) [inline lib/registries/addon/components/registries-navbar/component.ts:35]: Should we just go all the way with this and put it in settings? It feels like a thing that would be in settings.

**Review by brianjgeiger — CHANGES_REQUESTED** [review]
One suggestion, since we're trying to get rid of magic strings.

**Review by brianjgeiger — APPROVED** [review]
Nice

**Review by fabmiz — APPROVED** [review]
🥇

**Pull request merged**